portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -75,18 +80,39 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
         messages.add(message);
       }
 
+      // Create linkedHashMap and put in BlogMessages. 
+      Map<String, LinkedList<BlogMessage>> map = new LinkedHashMap<String, LinkedList<BlogMessage>>();
```

Comment:
This is why we want to make BlogHashMap a better abstraction. This right now basically exposes all internal details to the client code, which is particularly undesirable. A better abstraction would be that I can use the BlogHashMap like:

BlogHashMap blog_map = new BlogHashMap();
blog_map.putInMap(messages);

And as a client of your BLogHashMap, I don't and should not know anything about how you organize the internal data structure. A good standard is that I should not change a single line of my code if you decided that a hashmap is not great and you need to make an even fancier data structure.

Reply:
Understood

Reply:
Done!